### PR TITLE
[yeoman-generator] Adding options 

### PR DESCRIPTION
**Pull request by ErikElkins**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://yeoman.io/authoring/user-interactions.html
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: yeoman-generator/index.d.ts [hunk 1]
```diff
@@ -68,6 +68,7 @@ declare class Base extends EventEmitter {
     appname: string;
     config: Base.Storage;
     fs: Base.MemFsEditor;
+    options: any;
```
Comment (janslow) [inline yeoman-generator/index.d.ts]: `options` is always an object (same as `Base#args`) so it should be `{}` type.

**Review by janslow — CHANGES_REQUESTED** [review]

File: yeoman-generator/index.d.ts (hunk 1 above)
Comment (CaringIsCreepy) [inline yeoman-generator/index.d.ts]: Wouldn't that not compile in use in a generator though? It wouldn't be able to access a property like `this.options.debugFlag`

Comment (janslow) [inline yeoman-generator/index.d.ts]: You can make the type more specific in your class:

```typescript
class MyGenerator extends Base {
  options: {
    foo: boolean;
  };
}
```

But using `{}` prevents the following:

```
class MyGenerator extends Base {
  options: string; // options is never a string.
}
```

Comment (CaringIsCreepy) [inline yeoman-generator/index.d.ts]: Ah, that makes sense. I just updated it.

**Comment by janslow** [comment]
Ok, LGTM

**Review by janslow — APPROVED** [review]

**Pull request merged**